backend/models/submission.js

```diff
@@ -0,0 +1,81 @@
+const mongoose = require("mongoose");
+
+const matrix = new mongoose.Schema({
+  x: {
+    type: Number
+  },
+  y: {
+    type: Number
+  }
+});
+
+const option = new mongoose.Schema({
+  min: {
+    type: Number
+  },
+  max: {
+    type: Number
+  },
+  opt: {
+    type: String,
+    default: null
+  }
+});
+
+const answer = new mongoose.Schema({
+  type: {
+    type: String,
+    enum: [
+      "SHORT_ANSWER", //Yoptions.length = 0 | Xoptions.length = 1
+      "PARAGRAPHS", //Yoptions.length = 0 | Xoptions.length = 1
+      "MULTIPLE_CHOICE", //Yoptions.length = # of MC | Xoptions.length = 1
+      "CHECKBOXES", //Yoptions.length = # of boxes | Xoptions.length = 1
+      "FILE_UPLOAD", //Yoptions.length = 0 | Xoptions.length = 1
+      "CHECKBOX_GRID" //(Y|X)options.length = # of choices
+    ]
+  },
+  // SHORT_ANSWER
+  // PARAGRAPH
+  answerString: {
+    type: String
+  },
+  // MULTIPLE_CHOICE
+  // CHECKBOXES
+  answerArray: {
+    type: [option]
+  },
+  // CHECKBOX_GRID
+  answerMatrix: {
+    type: [matrix]
+  },
+  sectionId: {
+    type: String
+  },
+  questionId: {
+    type: String
+  }
+});
+
+const submissionsSchema = new mongoose.Schema(
+  {
+    formId: {
+      type: String
+    },
+    linkId: {
+      type: String
+    },
+    submissionDate: {
+      type: String,
```

Comment:
Should we use MongoDB's Date object? (see [Mongoose docs](https://mongoosejs.com/docs/tutorials/dates.html)). This could allow more consistent date formatting & easier date comparison.

Reply:
Great idea! Just updated the types for `submissionDate` and `lastSaveDate`.